### ISSUE TITLE
Unable to use dict union on typedicts

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#2239]
**Describe the bug**
3.9 added the [union operator to dicts](https://www.python.org/dev/peps/pep-0584/) which allows an easy way to join dicts together, when trying this on a typedict pyright errors saying its not supported.

**To Reproduce**
```py
class A(TypedDict):
    key: int

class B(TypedDict):
    key2: str

a: A = {"key": 1}
b: B = {"key2": ""}

reveal_type(a | b)
```
```
Operator "|" not supported for types "A" and "B"
```
**Expected behavior**
i expect the output to be a dict and to not error.

**Additional context**
In an ideal world i would love the output to be a intersection type of `A & B`, however i doubt intersection will ever happen https://github.com/python/typing/issues/213

**Comment by erictraut** [comment]
Pyright (and other Python type checkers like mypy) use the `_TypedDict` type declaration in `typing.pyi` to determine which operations are legal for TypedDict instances. Here's the relevant portion of that declaration:

```python
class _TypedDict(Mapping[str, object], metaclass=ABCMeta):
    ...
    def __or__(self: _T, __value: _T) -> _T: ...
    ...
```

Based on this declaration, the `|` operator works with two same-typed TypedDict instance but not with different-typed TypedDict instances. So Pyright is doing the correct thing here. It emits the same error as mypy in this case.

If you think this is an error, please file a bug in the [typeshed repo](https://github.com/python/typeshed). Pyright regularly pulls from typeshed, so if you convince the maintainers of typeshed to make this change, pyright will pick up the new declaration shortly afterward.
